lib/src/sql/duration.rs

```diff
@@ -257,22 +257,25 @@ pub fn duration(i: &str) -> IResult<&str, Duration> {
 fn duration_raw(i: &str) -> IResult<&str, Duration> {
 	let (i, v) = part(i)?;
 	let (i, u) = unit(i)?;
-	Ok((
-		i,
-		Duration(match u {
-			"ns" => time::Duration::from_nanos(v),
-			"µs" => time::Duration::from_micros(v),
-			"us" => time::Duration::from_micros(v),
-			"ms" => time::Duration::from_millis(v),
-			"s" => time::Duration::from_secs(v),
-			"m" => time::Duration::from_secs(v * SECONDS_PER_MINUTE),
-			"h" => time::Duration::from_secs(v * SECONDS_PER_HOUR),
-			"d" => time::Duration::from_secs(v * SECONDS_PER_DAY),
-			"w" => time::Duration::from_secs(v * SECONDS_PER_WEEK),
-			"y" => time::Duration::from_secs(v * SECONDS_PER_YEAR),
-			_ => time::Duration::ZERO,
-		}),
-	))
+
+	let std_duration = match u {
+		"ns" => Some(time::Duration::from_nanos(v)),
+		"µs" => Some(time::Duration::from_micros(v)),
+		"us" => Some(time::Duration::from_micros(v)),
+		"ms" => Some(time::Duration::from_millis(v)),
+		"s" => Some(time::Duration::from_secs(v)),
+		"m" => v.checked_mul(SECONDS_PER_MINUTE).map(time::Duration::from_secs),
+		"h" => v.checked_mul(SECONDS_PER_HOUR).map(time::Duration::from_secs),
+		"d" => v.checked_mul(SECONDS_PER_DAY).map(time::Duration::from_secs),
+		"w" => v.checked_mul(SECONDS_PER_WEEK).map(time::Duration::from_secs),
+		"y" => v.checked_mul(SECONDS_PER_YEAR).map(time::Duration::from_secs),
+		_ => {
+			debug_assert!(false, "shouldn't have parsed {u} as duration unit");
+			Some(time::Duration::ZERO)
+		}
```

Comment:
Fair enough...I'm just slightly paranoid about adding runtime errors in release mode :joy:

Reply:
Shall we do a separate PR and actually create a new `Error` type (`lib/src/err/mod.rs`) type for runtime errors, and replace most instances of `unreachable!()` in the query processing runtime with our custom error? We should not ever see that error, but if it does happen, then at least it isn't a panic! What are your thoughts?

Reply:
Sounds good to me. In cases like this, it is also possible to make a new `enum` type so that Rust can statically verify that the matched variants are exhaustive.

Reply:
That's a good idea too!